### PR TITLE
Improve Ratpack context propagation and migrate tests

### PR DESCRIPTION
**Pull request by tylerbenson**
Adds a "handler" span option to the base `HttpServerTest`.

### CODE REVIEW

File: dd-java-agent/instrumentation/ratpack-1.4/src/main/java/datadog/trace/instrumentation/ratpack/ContinuationInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        } else {
+          delegate.execute();
+        }
+      }
+
+      public static Block wrapIfNeeded(final Block delegate, final Span span) {
+        if (delegate instanceof BlockWrapper || span == null) {
+          return delegate;
+        }
+        log.debug("Wrapping action task {}", delegate);
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/ratpack-1.4/src/main/java/datadog/trace/instrumentation/ratpack/ContinuationInstrumentation.java]:
```suggestion
        log.debug("Wrapping block {}", delegate);
```

File: dd-java-agent/instrumentation/ratpack-1.4/src/main/java/datadog/trace/instrumentation/ratpack/ContinuationInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    }
+
+    public void muzzleCheck(final PathBinding binding) {
+      // This was added in 1.4.  Added here to ensure consistency with other instrumentation.
+      binding.getDescription();
+    }
+
+    @Slf4j
+    public static class BlockWrapper<T> implements Block {
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/ratpack-1.4/src/main/java/datadog/trace/instrumentation/ratpack/ContinuationInstrumentation.java]:
```suggestion
    public static class BlockWrapper implements Block {
```

File: dd-java-agent/instrumentation/ratpack-1.4/src/test/groovy/server/RatpackHttpServerTest.groovy [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  String expectedOperationName() {
+    "netty.request"
+  }
+
+  @Override
+  boolean hasHandlerSpan() {
+    true
+  }
+
+  void cleanAndAssertTraces(
```
[the remark this answers is not in the document]
Comment (tylerbenson) [reply dd-java-agent/instrumentation/ratpack-1.4/src/test/groovy/server/RatpackHttpServerTest.groovy]: good idea.  I'll just do it here.

[the remark this answers is not in the document]
Comment (tylerbenson) [reply dd-java-agent/instrumentation/ratpack-1.4/src/test/groovy/server/RatpackHttpServerTest.groovy]: good idea. I'll just do it here.

**Pull request merged**